diff --git a/glib/gdate.c b/glib/gdate.c
index 670f7aa..399f8d3 100644
--- a/glib/gdate.c
+++ b/glib/gdate.c
@@ -2439,6 +2439,8 @@ win32_strftime_helper (const GDate     *d,
  *
  * Returns: number of characters written to the buffer, or 0 the buffer was too small
  */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
 gsize     
 g_date_strftime (gchar       *s, 
                  gsize        slen, 
@@ -2549,3 +2551,4 @@ g_date_strftime (gchar       *s,
   return retval;
 #endif
 }
+#pragma GCC diagnostic pop
